lib/modules/versioning/rust/index.ts

```diff
@@ -0,0 +1,201 @@
+import { logger } from '../../../logger';
+import type { RangeStrategy } from '../../../types/versioning';
+import { regEx } from '../../../util/regex';
+import { api as npm } from '../npm';
+import type { NewValueConfig, VersioningApi } from '../types';
+
+export const id = 'rust';
+export const displayName = 'Rust';
+export const urls = [
+  'https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file',
+];
+
+export const supportsRanges = true;
+export const supportedRangeStrategies: RangeStrategy[] = [
+  'bump',
+  'pin',
+  'replace',
+];
+
+// Format described in https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file
+// Versions may be fully qualified: <major>.<minor>.<patch> (e.g. 1.82.1), indicating an exact version.
+// Or version may be inexact: <major>.<minor> (e.g. 1.82), allowing patch level changes.
```

Comment:
Where did you find the inexact definition? On that page I see the definition as:

```
<channel>[-<date>]

<channel>       = stable|beta|nightly|<major.minor.patch>
<date>          = YYYY-MM-DD
```

This doesn't indicate patch as optional